mapbox_vector_tile/decoder.py

```diff
@@ -109,8 +119,16 @@ def parse_geometry(self, geom, ftype):
                     dx = x
                     dy = y
 
-                    coords.append([x, 4096-y])
-
-        if ftype == POLYGON and len(coords) > 0:
-            coords.append(coords[0])
-        return coords
+                    coords.append([x, self.extents-y])
+
+        if ftype == POINT:
+            return coords
+        elif ftype == LINESTRING or ftype == POLYGON:
+            if parts:
+                if coords:
+                    parts.append(coords)
+                return parts[0] if len(parts) == 1 else parts
```

Comment:
Yup, absolutely. I saw that it was being [assigned as-is](https://github.com/mapzen/mapbox-vector-tile/blob/22-update-multipolygon-behavior/mapbox_vector_tile/decoder.py#L50) to the `geometry` field of the returned feature and assumed it would get passed directly to Shapely at some point. I think it makes sense to be explicit about what this function returns and where the responsibility for turning `ftype=POLYGON` into Polygon / MultiPolygon lies. The effect of the spec is that, until we do an area calculation, we don't know whether we've got one or the other.